open_spiel/games/battleship.h

```diff
@@ -0,0 +1,296 @@
+//                                     player                 (default = 2)
+//     "allow_repeated_shots" bool     If false, the players will be prevented
+//                                     from shooting multiple times at the same
+//                                     cell of the board      (default = false)
+//     "loss_multiplier"    double     Loss multiplier (see above). The game is
+//                                     zero-sum iff the loss multiplier is 1.0
+//                                                            (default = 2.0)
+//
+// NOTE: The list parameters must be supplied as a string of comma-separated
+// values
+//
+// References
+// ==========
+//
+// [1]:
+// https://papers.nips.cc/paper/9122-correlation-in-extensive-form-games-saddle-point-formulation-and-benchmarks.pdf
+//
+// If you want to reference the paper that introduced the benchmark game, here
+// is a Bibtex citation:
+//
+// ```
+// @inproceedings{Farina19:Correlation,
+//    title=    {Correlation in Extensive-Form Games: Saddle-Point Formulation
+//               and Benchmarks},
+//    author=   {Farina, Gabriele and Ling, Chun Kai and Fang, Fei and
+//               Sandholm, Tuomas},
+//    booktitle={Conference on Neural Information Processing Systems
+//               (NeurIPS)},
+//    year={2019}
+// }
+// ```
+
+#ifndef OPEN_SPIEL_GAMES_BATTLESHIP_H_
+#define OPEN_SPIEL_GAMES_BATTLESHIP_H_
+
+#include <memory>
+#include <vector>
+
+#include "open_spiel/games/battleship_types.h"
+#include "open_spiel/spiel.h"
+
+namespace open_spiel {
+namespace battleship {
+
+constexpr int kDefaultBoardWidth = 2;
+constexpr int kDefaultBoardHeight = 2;
+constexpr const char* kDefaultShipSizes = "2";
+constexpr const char* kDefaultShipValues = "1.0";
+constexpr int kDefaultNumShots = 2;
+constexpr bool kDefaultAllowRepeatedShots = false;
+constexpr double kDefaultLossMultiplier = 2.0;
+
+class BattleshipGame;
+class BattleshipState final : public State {
+ public:
+  explicit BattleshipState(const std::shared_ptr<const BattleshipGame> bs_game);
+  ~BattleshipState() = default;
+
+  // Virtual functions inherited by OpenSpiel's `State` interface
+  Player CurrentPlayer() const override;
+  std::vector<Action> LegalActions() const override;
+  std::string ActionToString(Player player, Action action) const override;
+  std::string ToString() const override;
+  bool IsTerminal() const override;
+  std::vector<double> Returns() const override;
+  std::unique_ptr<State> Clone() const override;
+  std::string InformationStateString(Player player) const override;
+  std::string ObservationString(Player player) const override;
+  void UndoAction(Player player, Action action) override;
+
+ protected:
+  void DoApplyAction(Action action) override;
+
+ private:
+  // Ship placement methods
+  // ======================
+
+  // Returns how many ships have been placed collectively by the two player.
+  int NumShipsPlaced_() const;
```

Comment:
We don't use this naming convention for private methods. I slightly prefer not having the underscores to keep it consistent with the other games.

Reply:
Ah yeah that makes a lot of sense! I forgot that the Google style guide only has underscores for private members but not private methods :-) I will remove the underscores from the method!